collector/consumer/processor/aggregateprocessor/internal/label_key.go

```diff
@@ -0,0 +1,73 @@
+package internal
+
+import (
+	"github.com/Kindling-project/kindling/collector/model"
+	"strconv"
+)
+
+type vType string
+
+const (
+	BooleanType vType = "boolean"
+	StringType  vType = "string"
+	IntType     vType = "int"
+)
+
+type LabelSelectors struct {
+	// Name: VType
+	selectors []LabelSelector
+}
+type LabelSelector struct {
+	Name  string
+	VType vType
+}
+
+func NewLabelSelectors(selectors ...LabelSelector) *LabelSelectors {
+	return &LabelSelectors{selectors: selectors}
+}
+
+func (s *LabelSelectors) GetLabelKeys(labels *model.AttributeMap) *LabelKeys {
+	keys := &LabelKeys{}
+	for i, selector := range s.selectors {
+		keys.keys[i].Name = selector.Name
+		keys.keys[i].VType = selector.VType
+		switch selector.VType {
+		case BooleanType:
+			keys.keys[i].Value = strconv.FormatBool(labels.GetBoolValue(selector.Name))
+		case StringType:
+			keys.keys[i].Value = labels.GetStringValue(selector.Name)
+		case IntType:
+			keys.keys[i].Value = strconv.FormatInt(labels.GetIntValue(selector.Name), 10)
+		}
+	}
+	return keys
+}
+
+type LabelKeys struct {
+	// LabelKeys will be used as key of map, so it is must be an array instead of a slice.
+	// Now 34 is enough for all cases. If there is more than 34 labels, must increase this value.
+	keys [34]LabelKey
+}
+
+type LabelKey struct {
+	Name  string
+	Value string
+	VType vType
+}
+
+func (k *LabelKeys) GetLabels() *model.AttributeMap {
+	ret := model.NewAttributeMap()
+	for _, key := range k.keys {
+		switch key.VType {
+		case BooleanType:
+			value, _ := strconv.ParseBool(key.Value)
+			ret.AddBoolValue(key.Name, value)
+		case StringType:
+			ret.AddStringValue(key.Name, key.Value)
+		case IntType:
+			value, _ := strconv.ParseInt(key.Value, 10, 64)
+			ret.AddIntValue(key.Name, value)
+		}
+	}
```

Comment:
I think using the `for i` instead of `for range` can get better performance here

Reply:
Actually, they perform nearly identically.

Reply:
Right, here `k.keys` is an array. Iterating an array using `for range value` will copy the original value. But since this method is called less frequently, the impact on performance is much little.